images/mediatypes.go

```diff
@@ -57,6 +57,12 @@ const (
 
 	MediaTypeImageLayerEncrypted     = ocispec.MediaTypeImageLayer + "+encrypted"
 	MediaTypeImageLayerGzipEncrypted = ocispec.MediaTypeImageLayerGzip + "+encrypted"
+
+	// OCI media types
+
+	MediaTypeWasmLayerModule    = "application/vnd.w3c.wasm.module.v1+wasm"
```

Comment:
Is this media type approved by W3C?
Ofherwise “vnd.w3c” doesn’t seem correct.

Reply:
These types are not set in stone.  I got the from https://hackmd.io/50rfwV6BTJWN8VZBhdAN_g and https://github.com/bytecodealliance/registry/pull/87.  

I needed something to test with but I need some input on long term types and best way to have a type that we can begin using in containerd and testing to prove out this work.  

One option I considered was to not have any types in containerd and have the `WithWasmLayers` be `WithOciArtifact` and it would just add any layers it didn't recognize but not sure I'd adding anything at all os a good idea.

Reply:
There is a wasm type registered in https://www.iana.org/assignments/media-types/media-types.xhtml

Reply:
Maybe we should do `vnd.containerd...`?
I suspect this will need to converge on a more generic name soon (especially after we introduce something) but it seems wrong to create one using w3c's name here.

Reply:
Agree. `vnd.containerd` is interesting idea.  

I would like to be in agreement with what lands in https://github.com/bytecodealliance/registry/pull/87, so we don't have to there aren't to many changes on containerd side.  

Any thoughts on making it more generic in the time being? Something like `WithOciArtifact` so it would add any layers it didn't recognize to the annotation?

Reply:
Oh maybe a `vnd.bytecodealliance` could work?

Reply:
Just that it could end up blocking progress.

Reply:
Hi folks. I got pinged... 
For defining unique types, there documentation under: [OCI/Artifacts: Defining a Unique Artifact Type](https://github.com/opencontainers/artifacts/blob/main/artifact-authors.md#defining-a-unique-artifact-type) that might be helpful. It explains `[registration-tree]`.`[org|company|entity]`.`[objectType]`.`[optional-subType]`.config.`[version]`+`[optional-configFormat]`